### PR TITLE
[NEXUS-8410] use configfacet bean mapping to read config

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8410

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java [hunk 1]
```diff
@@ -44,11 +44,26 @@
...
 
-  private final Set<String> memberNames = Sets.newLinkedHashSet();
+  @VisibleForTesting
+  static final String CONFIG_KEY = "group";
+
+  @VisibleForTesting
+  static class Config
+  {
+    @NotEmpty
+    Set<String> memberNames;
```
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java]: We many need to use List here, or sort out how jackson can be taught to retain set ordering.

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF167-1 :white_check_mark:

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**